### PR TITLE
feat(core-p2p): hapi/nes implementation

### PR DESCRIPTION
**Pull request by air1one**
## Summary

WIP on hapi/nes PoC implementation of the websocket layer (instead of socketcluster).

## Checklist

- [ ] Documentation _(if necessary)_
- [ ] Tests _(if necessary)_
- [ ] Ready to be merged

### CODE REVIEW

File: packages/core-p2p/src/socket-server/controllers/peer.ts [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+
+        const blockchain = this.app.get<Contracts.Blockchain.Blockchain>(Container.Identifiers.BlockchainService);
+        return {
+            common: commonBlocks[0],
+            lastBlockHeight: blockchain.getLastBlock().data.height,
+        };
+    }
+
+    public async getStatus(request: Hapi.Request, h: Hapi.ResponseToolkit): Promise<Contracts.P2P.PeerPingResponse> {
+        this.validatePayload(request.payload, peerSchemas.getStatus);
```
Comment (faustbrian) [inline packages/core-p2p/src/socket-server/controllers/peer.ts:67]: Could this be moved into a plugin that is responsible for validation? Seems to be duplicated in every controller method.

File: packages/core-p2p/src/socket-server/controllers/peer.ts [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+    }
+
+    public async getBlocks(
+        request: Hapi.Request,
+        h: Hapi.ResponseToolkit,
+    ): Promise<Interfaces.IBlockData[] | Contracts.Shared.DownloadBlock[]> {
+        this.validatePayload(request.payload, peerSchemas.getBlocks);
+        
+        // Add the peer if not already (on every peer endpoint)
+        this.peerProcessor.validateAndAcceptPeer({ ip: request.info.remoteAddress } as Contracts.P2P.Peer);
```
Comment (faustbrian) [inline packages/core-p2p/src/socket-server/controllers/peer.ts:180]: Could this be moved into a plugin that is responsible for peer acceptance? Seems to be duplicated in every controller method.

Comment (air1one) [reply packages/core-p2p/src/socket-server/controllers/peer.ts]: done.

File: packages/core-p2p/src/socket-server/controllers/peer.ts (hunk 1 above)
Comment (air1one) [reply packages/core-p2p/src/socket-server/controllers/peer.ts]: done.

**Comment by faustbrian** [comment]
This line https://github.com/ArkEcosystem/core/blob/p2p/hapi-nes/jest.config.js#L21 needs to be removed now that the initial rework was done so we can make sure we get back to 100% coverage for it.

**Pull request merged**